### DISCUSSION TITLE
Detect a "DRAFT" watermark?

### DISCUSSION THREAD

**Discussion by mnoah66** [pymupdf/PyMuPDF#1859] [Looking for help] [answered]
Is it possible to just detect if "DRAFT" watermark exists?  I have seen some google search results that want to remove or replace it.  I want neither. 

Just detect :)

I've tried drawing rectangle and getting text:
```
draft = fitz.Rect(65, 120, 520,  745)
print("DRAFT" in page.get_textbox(draft)) # False
```

[PDF image](https://i.imgur.com/JNnMnav.png). The pdf has medical info and cannot be shared

**Comment by JorjMcKie** [comment, marked as answer]
Hard to give a definitive answer if you cannot share an example file. But maybe your watermark can be removed by the recipe I developed [here](https://github.com/pymupdf/PyMuPDF/discussions/1855).

**Reply by mnoah66** [reply]
I used the [extract-imga.py](https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/examples/extract-imga.py) script which resulted in a few images, one of which being the "DRAFT" watermark.  So now, I presume I'll need to loop through the images, get_text, and check for "DRAFT"

https://i.imgur.com/bu2e6av.png

Thank you!

**Reply by JorjMcKie** [reply]
I recommend you try this script first before you invest too much in other things:
[remove-watermarks.zip](https://github.com/pymupdf/PyMuPDF/files/9269490/remove-watermarks.zip)

**Reply by mnoah66** [reply]
I ran that script and it resulted in "Nothing to change"

**Reply by JorjMcKie** [reply]
Interesting.
Then maybe the image is inserted on the page "traditionally" - i.e. not using PDF watermark technology.
Depending on how you extracted the image, xref 3 may contain it. Check this by `print(page.get_images())`.
If so, then you can simply replace the image by a small invisible other image using [this](https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/image-replacement/remover.py) example script.

You have no chance to share an example file? Even to my personal e-mail address?

**Reply by mnoah66** [reply]
I emailed you the file.  I am not looking to replace the image/watermark.  Just detect if it is there.

Thank you! :)

````
print(page.get_images())
[(3, 0, 816, 912, 8, 'DeviceRGB', '', 'Im3', 'FlateDecode'), (5, 0, 33, 83, 8, 'DeviceRGB', '', 'Im5', 'FlateDecode'), (7, 0, 130, 128, 8, 'DeviceRGB', '', 'Im7', 'FlateDecode'), (9, 0, 1329, 106, 8, 'Indexed', '', 'Im9', 'FlateDecode'), (10, 0, 916, 463, 8, 'DeviceRGB', '', 'Im10', 'FlateDecode')]

```

**Reply by JorjMcKie** [reply]
Thanks for the file.
As suspected, this is no watermark - just a transparent image. As such, you cannot definitely tell what its role is - you know nothing about it except the few metadata contained in dictionary `doc.extract_image(3)`.
You can also determine, where in which location on which page this image is being shown.
That's about it. Strictly speaking, **_this is no watermark._**

**Reply by mnoah66** [reply]
@JorjMcKie thanks again.  I think I have enough to work with given that it is the only image with a height of over 900.  I could loop over the images and check for the height and deduce if it is a draft.  I think adding OCR would be overkill.

**Reply by JorjMcKie** [reply]
>  I think adding OCR would be overkill.

I tried that already. Neither Tesseract, nor OCRmyPDF (which is based on that anyway) can decipher that image.
